packages/core/src/StateMachine.ts

```diff
@@ -307,13 +313,21 @@ export class StateMachine<
     return transitionNode(this.root, state.value, state, _event) || [];
   }
 
+  public preInitialState:
+    | State<TContext, TEvent, TResolvedTypesMeta>
+    | undefined;
+
   /**
    * The initial state _before_ evaluating any microsteps.
    * This "pre-initial" state is provided to initial actions executed in the initial state.
    */
   private getPreInitialState(
     actorCtx: ActorContext<any, any> | undefined
   ): State<TContext, TEvent, TResolvedTypesMeta> {
+    if (this.preInitialState) {
+      return this.preInitialState;
+    }
```

Comment:
why do we have to cache it now?

Reply:
I don't remember but tests fail if I remove this part

Reply:
Removed